addons/mail/static/src/core/common/thread_model.js

```diff
@@ -168,23 +137,61 @@ export class Thread extends Record {
         },
     });
     invitedMembers = Record.many("ChannelMember");
-    chatPartner = Record.one("Persona");
-    composer = Record.one("Composer", { inverse: "thread", onDelete: (r) => r.delete() });
+    chatPartner = Record.one("Persona", {
+        /** @this {import("models").Thread} */
+        compute() {
+            if (this.type !== "chat") {
+                return undefined;
+            }
+            for (const member of this.channelMembers) {
+                if (
+                    !member.persona?.eq(this._store.user) ||
+                    (this.channelMembers.length === 1 && member.persona?.eq(this._store.user))
+                ) {
+                    return member.persona;
+                }
+            }
+        },
+    });
+    composer = Record.one("Composer", {
+        compute: () => ({}),
+        inverse: "thread",
+        onDelete: (r) => r.delete(),
+    });
     counter = 0;
     /** @type {string} */
     custom_channel_name;
     /** @type {string} */
     description;
     followers = Record.many("Follower");
-    selfFollower = Record.one("Follower");
+    selfFollower = Record.one("Follower", {
+        /** @this {import("models").Thread} */
+        onAdd(r) {
+            r.followedThread = this;
+        },
+        onDelete: (r) => (r.followedThread = undefined),
+    });
```

Comment:
Why is it not just an inverse of this field?

Reply:
indeed, could be an inverse